src/dex/ks-elastic/ks-elastic-pool.ts

```diff
@@ -0,0 +1,429 @@
+    } catch (e) {
+      this.logger.error(
+        `Error_${this.parentName}_processLog could not parse the log with topic ${log.topics}:`,
+        e,
+      );
+      return null;
+    }
+  }
+
+  getPoolStateRequest() {
+    const data = {
+      funcName: 'getPoolState',
+      params: [],
+    };
+    return data;
+  }
+
+  getPoolCallData(): stateRequestCallData {
+    const data = {
+      funcName: 'getPool',
+      params: [this.token0, this.token1, Number(this.fee)],
+    };
+    return data;
+  }
+
+  generateMultiCallInput(ticks: Number[]): multiCallInput[] {
+    return ticks.map(tickIndex => ({
+      target: this.poolAddress,
+      callData: this.poolIface.encodeFunctionData('ticks', [tickIndex]),
+    }));
+  }
+
+  decodeMultiCallResult(multiCallTickResult: []) {
+    const result = new Array(multiCallTickResult.length);
+    multiCallTickResult.forEach((element, index) => {
+      result[index] = this.poolIface.decodeFunctionResult('ticks', element);
+    });
+    return result;
+  }
+
+  private setTicksMapping(
+    ticks: Record<NumberAsString, TickInfo>,
+    tickArray: number[],
+    tickInfosFromContract: any[],
+  ) {
+    return tickInfosFromContract.reduce<Record<string, TickInfo>>(
+      (acc, element, index) => {
+        acc[tickArray[index]] = {
+          liquidityGross: bigIntify(element.liquidityGross),
+          liquidityNet: bigIntify(element.liquidityNet),
+          tickCumulativeOutside: bigIntify(element.feeGrowthOutside),
+          secondsPerLiquidityOutsideX128: bigIntify(
+            element.secondsPerLiquidityOutside,
+          ),
+          secondsOutside: bigIntify(
+            element.liquidityNet * element.secondsPerLiquidityOutside,
+          ),
+          initialized: true,
+          index: tickArray[index],
+        };
+        return acc;
+      },
+      ticks,
+    );
+  }
+
+  getAllTickRequest(poolAddress: string): stateRequestCallData {
+    const data = {
+      funcName: 'getAllTicks',
+      params: [poolAddress],
+    };
+    return data;
+  }
+
+  async gePoolContract() {
+    try {
+      const getPoolData = this.getPoolCallData();
+      const poolAddress = await this.poolFactoryContract.methods[
+        getPoolData.funcName
+      ](...getPoolData.params).call();
+      if (poolAddress === PoolAddressNotExisted) {
+        throw new PoolNotFoundError(this.token0, this.token1, this.fee);
+      }
+      this.poolAddress = poolAddress;
+      return new this.dexHelper.web3Provider.eth.Contract(
+        PoolInstanceABI as AbiItem[],
+        poolAddress,
+      );
+    } catch (error) {
+      throw error;
+    }
+  }
+
+  getPoolState(blockNumber: number) {
+    const callRequest = this.getPoolStateRequest();
+    return this.poolContract.methods[callRequest.funcName](
+      ...callRequest.params,
+    ).call({}, blockNumber || 'latest');
+  }
+
+  getAllTick(poolAddress: string, blockNumber: number) {
+    const callRequest = this.getAllTickRequest(poolAddress);
```

Comment:
This function returns an out of gas error when running for top pools (I am assuming with lots of ticks). e.g. pool with address `0x087abaab9cd85025a8b3916948c69fe173c837ea` on Arbitrum